pkg/storage/replica.go

```diff
@@ -1143,15 +1144,104 @@ func (r *Replica) checkBatchRequest(ba roachpb.BatchRequest) error {
 // error is to be used in place of the supplied error.
 func (r *Replica) beginCmds(
 	ctx context.Context, ba *roachpb.BatchRequest,
-) (func(*roachpb.BatchResponse, *roachpb.Error) *roachpb.Error, error) {
+) (func(*roachpb.BatchResponse, *roachpb.Error, interface{}) *roachpb.Error, error) {
 	var cmd *cmd
+	var spans []roachpb.Span
 	// Don't use the command queue for inconsistent reads.
 	if ba.ReadConsistency != roachpb.INCONSISTENT {
 		readOnly := ba.IsReadOnly()
-		spans := make([]roachpb.Span, len(ba.Requests))
+		spans = make([]roachpb.Span, len(ba.Requests)+1)
+		spans[len(ba.Requests)] = roachpb.Span{
+			Key: keys.RangeFrozenStatusKey(r.RangeID),
+		}
+		// TODO(tschottdorf): with all of these things added to the spans,
+		// the heuristic we are using in the CommandQueue to make a large
+		// span first is going to be horrible as we are always including
+		// range-local and range-global keys, and so that mutant span includes
+		// the whole world.
```

Comment:
Agreed. We'll want to be much more fine grained about what's read-only and what's read-write (much of the system spans are almost always accessed read-only).